openjdk/src/main/java/org/conscrypt/Java8PlatformUtil.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.conscrypt;
+
+import static javax.net.ssl.StandardConstants.SNI_HOST_NAME;
+
+import java.util.Collections;
+import java.util.List;
+import javax.net.ssl.SNIHostName;
+import javax.net.ssl.SNIServerName;
+import javax.net.ssl.SSLParameters;
+
+/**
+ * Utility methods supported on Java 8+.
+ */
+final class Java8PlatformUtil {
```

Comment:
Won't this class just fail to compile under Java 6, since it uses a bunch of classes that don't exist?  I don't see anywhere that we exclude it from the build when we're compiling using previous versions.  Or are we assuming that we can compile using Java 8, and just avoid loading this class at runtime when running under previous versions?

Basically, we should be more explicit about what we mean by "support Java 6+": compiles using that version, can be compiled into classfiles that work under that version, only uses language-level features from that version, etc.